rct229/rules/section23/section23rule8.py

```diff
@@ -0,0 +1,137 @@
+
+    def list_filter(self, context_item, data):
+        hvac_sys_b = context_item.baseline
+        applicable_hvac_sys_ids = data["applicable_hvac_sys_ids"]
+
+        return hvac_sys_b["id"] in applicable_hvac_sys_ids
+
+    class HVACRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section23Rule8.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section23Rule8.HVACRule.SupplyFanRule(),
+                index_rmr="baseline",
+                list_path="$..supply_fans[*]",
+            )
+
+        class SupplyFanRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section23Rule8.HVACRule.SupplyFanRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    required_fields={
+                        "$": [
+                            "design_airflow",
+                            "design_electric_power",
+                            "output_validation_points",
+                        ],
+                    },
+                )
+
+            def get_calc_vals(self, context, data=None):
+                supply_fan_b = context.baseline
+
+                design_airflow_b = supply_fan_b["design_airflow"]
+                design_electric_power_b = supply_fan_b["design_electric_power"]
+                output_validation_points_b = supply_fan_b["output_validation_points"]
+
+                output_validation_points = [
+                    [output["airflow"], output["result"]]
+                    for output in output_validation_points_b
+                ]
+
+                target_validation_points = [
+                    [
+                        SUPPLY_AIRFLOW_COEFF[idx] * design_airflow_b,
+                        DESIGN_POWER_COEFF[idx] * design_electric_power_b,
+                    ]
+                    for idx in range(VALIDATION_POINTS_LENGTH)
+                ]
+
+                return {
+                    "design_airflow_b": CalcQ("air_flow_rate", design_airflow_b),
+                    "design_electric_power_b": CalcQ(
+                        "electric_power", design_electric_power_b
+                    ),
+                    "output_validation_points": output_validation_points,
+                    "target_validation_points": target_validation_points,
+                }
+
+            def rule_check(self, context, calc_vals=None, data=None):
+                output_validation_points = calc_vals["output_validation_points"]
+                target_validation_points = calc_vals["target_validation_points"]
+
+                return len(
+                    output_validation_points
+                ) == VALIDATION_POINTS_LENGTH and all(
+                    [
+                        std_equal(x[0], y[0]) and std_equal(x[1], y[1])
+                        for x, y in zip(
```

Comment:
Maybe use variables `ovp` and `tvp` instead of the more generic `x` and `y`?

Reply:
Fixed.